File: application/src/java/wandledi/example/models/Comment.java
package wandledi.example.models;

import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.Lob;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

/**
 *
 * @author Markus Kahl
 */
@Entity
public class Comment {

    private String author;
    private String email;

    @Lob
    private String content;
    @Temporal(TemporalType.TIMESTAMP)
    private Date date;

    public Comment() {

        date = new Date();
    }

    /**
     * @return the author
     */
    public String getAuthor() {
        return author;
    }

    /**
     * @param author the author to set
     */
    public void setAuthor(String author) {
        this.author = author;
    }

    /**
     * @return the email
     */
    public String getEmail() {
        return email;
    }

    /**
     * @param email the email to set
     */
    public void setEmail(String email) {
        this.email = email;
    }

    /**
     * @return the content
     */
    public String getContent() {
        return content;
    }

    /**
     * @param content the content to set
     */
    public void setContent(String content) {
        this.content = content;
    }

    /**
     * @return the date
     */
    public Date getDate() {
        return date;
    }

    /**
     * @param date the date to set
     */
    public void setDate(Date date) {
        this.date = date;
    }
}

File: scala-app/src/main/scala/wandledi/scala/example/models/Comment.scala
package wandledi.scala.example.models

import scala.reflect.BeanProperty
import wandledi.scala.Entity
import wandledi.scala.EntityCompanion
import wandledi.scala.Parameters
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.Lob
import javax.persistence.Temporal
import javax.persistence.TemporalType

@javax.persistence.Entity
class Comment(commentAuthor: String, commentEmail: String, commentContent: String) extends Entity {

  @Id
  @GeneratedValue
  @BeanProperty
  var id: Long = _

  @BeanProperty
  var author: String = commentAuthor
  @BeanProperty
  var email: String = commentEmail

  @Lob
  @BeanProperty
  var content: String = commentContent

  @Temporal(TemporalType.TIMESTAMP)
  @BeanProperty
  var date: java.util.Date = new java.util.Date

  def this() = this(null, null, null)

  def this(params: Parameters) = this(params("author"), params("email"), params("content"))

  def validate: Boolean = ok(author) && (!ok(email) || email.contains("@")) && ok(content)

  def ok(field: String): Boolean = field != null && !field.isEmpty
}

object Comment extends EntityCompanion[Comment] {

  override def entityClass = classOf[Comment]
}
